controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -391,3 +380,12 @@ func checkStaticPodReadyCondition(pod *corev1.Pod) error {
 	}
 	return nil
 }
+
+func firstNodeNotMatchingName(name string, nodes []corev1.Node) *corev1.Node {
```

Comment:
An alternative is to fold this into `removeMemberForNode`. I suggest this as `removeMemberForNode` is the only place this code is used, factoring this function out causes us to have to jump around to an additional function while reading `removeMemberForNode`, and since the code within this function is not particularly complex, I'm not sure it needs a test case/its own function. 

However, these reasons can all be argued against so I recognize this is a stylistic choice and you are free to ignore this comment and change nothing.

Reply:
I split it out mainly to unit test it.

Reply:
With the `firstNodeNotMatchingName` code in-lined, I found `removeMemberForNode` hard to read. I'll keep as-is for now.